packages/datetime2/src/components/date-input2/dateInput2.tsx

```diff
@@ -570,40 +571,75 @@ export const DateInput2: React.FC<DateInput2Props> = React.memo(function _DateIn
     const shouldShowErrorStyling =
         !isInputFocused || inputValue === props.outOfRangeMessage || inputValue === props.invalidDateMessage;
 
+    // We use the renderTarget API to flatten the rendered DOM and make it easier to implement features like
+    // the "fill" prop. Note that we must take `isOpen` as an argument to force this render function to be called
```

Comment:
this copy-pasted comment is a little wrong, the `isOpen` part is no longer necessary to point out where we have a dependency list (compared to the class component where this comment came from)